stdlib/traceback.pyi

```diff
@@ -85,7 +85,21 @@ def format_list(extracted_list: Iterable[FrameSummary | _FrameSummaryTuple]) ->
 # undocumented
 def print_list(extracted_list: Iterable[FrameSummary | _FrameSummaryTuple], file: SupportsWrite[str] | None = None) -> None: ...
 
-if sys.version_info >= (3, 10):
+if sys.version_info >= (3, 13):
+    @overload
+    def format_exception_only(exc: BaseExceptionGroup[BaseException], /, *, show_group: bool = False) -> list[str]: ...
+    @overload
+    def format_exception_only(
+        exc: Unused, /, value: BaseExceptionGroup[BaseException], *, show_group: bool = False
+    ) -> list[str]: ...
+    @overload
+    def format_exception_only(exc: BaseException | None, /, *, show_group: Literal[False] = False) -> list[str]: ...
+    @overload
+    def format_exception_only(
+        exc: Unused, /, value: BaseException | None, *, show_group: Literal[False] = False
+    ) -> list[str]: ...
```

Comment:
I don't think we need - or should - special case `show_group`. If `show_group` is `True`, but the exception is not an exception group, it will be printed as normal. If it's `False`, but it is an exception group, only a single exception will be printed - and that is fine. I think it was a deliberate design decision that this works either way, but cc @iritkatriel.

Also, `format_exception_only` now has a `colorize` option, defaulting to `False`, which has weirdly being hidden in the `kwargs`.

Reply:
I agree we don't _need_ to special-case `show_group`. I added some special-casing because it seemed to me like it was a case similar to others in typeshed, where we alert users to code that wouldn't necessarily generate an exception at runtime, but would be useless. AFAIK, there's no possible reason to pass `show_group=True` if the object being passed in is not known to be an `ExceptionGroup` instance, so it seems indicative of a probable bug if a user does that -- therefore something that a type checker could usefully complain about.

Having said all that, however, I also wasn't sure that this was worth the added complexity here, so I'll remove the special-casing for `show_group`.

> Also, `format_exception_only` now has a `colorize` option, defaulting to `False`, which has weirdly being hidden in the `kwargs`.

Yes, but I discussed in the PR description why I was deliberately not adding the `_colorize` or `colorize` keyword arguments to any of the `traceback` functions in this PR. See e.g. https://github.com/python/cpython/pull/111567#discussion_r1590033746

Reply:
> AFAIK, there's no possible reason to pass `show_group=True` if the object being passed in is not known to be an `ExceptionGroup` instance, so it seems indicative of a probable bug if a user does that -- therefore something that a type checker could usefully complain about.

I have to admit that my understanding is limited, but I think the point is that if you have a `BaseException` you don't know whether it's a group or not. And maybe you want to format all group members in case it's a group:

```python
try:
    ...  # might raise an exception group or single exceptions
except BaseException as exc:
    format_exception_group(exc, show_group=True)
```

Reply:
yeah, that's a fair point